scene/animation/tween.cpp

```diff
@@ -709,10 +709,11 @@ void Tween::_tween_process(float p_delta) {
 		InterpolateData &data = E->get();
 
 		// Track if we hit one that isn't finished yet
-		all_finished = all_finished && data.finish;
+		all_finished = all_finished && (!data.active || data.finish);
 
 		// Is the data not active or already finished? No need to go any further
 		if (!data.active || data.finish) {
+			all_finished = true;
```

Comment:
It still doesn't seem correct to me. Read the comments and the code again and this about how this algorithm is meant to work:
- `all_finished` is assumed true initially.
- For each tween, we check if it's either inactive (`!data.active`) or finished (`data.finish`) to confirm if `all_finished` is still true (if the tween is active or not finished, then `all_finished` **must become `false`**.
- If the current tween is finished, we skip to the next.

Your current code will make it so that as regardless of previous values of `all_finished`, the variable with be set to `true` any time the iterated tween is finished (so missed the point of the `all_` in the variable name).

Unroll the code on paper if it helps. For the sake of simplicity, let's assumed "finished" means `!data.active || data.finish`, so either `!active` or `finish` is true.
Here's what your current PR does:

| Step | Tween state | `all_finished` |
| ----- | ----- | ----- |
| Before loop | -- | `true` |
| Tween 0 | Finished | `true` |
| Tween 1 | Not finished | `false` |
| Tween 2 | Finished | `true` |
| After loop | --- | `true` |

The Tween will be deactivated and emit `tween_all_completed` even though Tween 1 is not finished.